docs/validation.md

```diff
@@ -434,130 +504,221 @@ author.createField('fullName')
+  .type('ResourceLink')
+  .allowedResources([
+    {
+      type: 'InvalidType'
+      // ...
+    }
+  ])
+```
+
+## field.allowedResources.TOO_FEW_ITEMS
+
+When trying to set an empty list of allowed resources.
+
+**Example:**
+
+```javascript
+author.createField('works').name('Works').type('ResourceLink').allowedResources([])
+```
+
+## field.allowedResources.TOO_MANY_ITEMS
+
+When trying to set more than 3 allowed resources on a field.
+
+**Example:**
+
+```javascript
+author.createField('works').name('Works').type('ResourceLink').allowedResources([
+  {
+    /* ... */
+  },
+  {
+    /* ... */
+  },
+  {
+    /* ... */
+  },
+  {
+    /* ... */
+  }
+])
+```
+
 ## DUPLICATE_PROP
+
 When setting the same property on a content type or field multiple times.
 
 **Example:**
+
 ```javascript
-author.createField('fullName')
-  .name('Full name')
-  .type('Symbol')
-  .name('Name')
+author.createField('fullName').name('Full name').type('Symbol').name('Name')
 ```
 
 Note that steps in the migration description are grouped into what will be requests to Contentful's API. If the property changes would not end up in the same request because they are done in a later, separate step, this will not produce an error.
 
 **Example:**
+
 ```javascript
-author.createField('fullName')
-  .name('Full name')
-  .type('Symbol')
+author.createField('fullName').name('Full name').type('Symbol')
 
-migration.deleteContentType('reader');
+migration.deleteContentType('reader')
 
-author.editField('fullName')
-  .name('Name')
+author.editField('fullName').name('Name')
 ```
 
 ## editorLayout.TOO_MANY_TABS
+
 When saving an editor layout with more than 5 tabs
 
 **Example:**
+
 ```javascript
 editorLayout.createFieldGroup('content', {
   name: 'Content'
-});
+})
 
 editorLayout.createFieldGroup('references', {
   name: 'References'
-});
+})
 
 editorLayout.createFieldGroup('settings', {
   name: 'Settings'
-});
+})
 
 editorLayout.createFieldGroup('seo', {
   name: 'SEO'
-});
+})
 
 editorLayout.createFieldGroup('keywords', {
   name: 'Keywords'
-});
+})
 
 editorLayout.createFieldGroup('extras', {
   name: 'Extras'
-});
-
+})
 ```
 
 ## editorLayout.TAB_CONTROL_INVALID
+
 When saving an editor layout with a tab that has `fieldset` control
 
 **Example:**
+
 ```javascript
 editorLayout.createFieldGroup('settings', {
   name: 'Settings'
-});
+})
 
-editorLayout.changeFieldGroupControl('settings', 'builtin', 'fieldset');
+editorLayout.changeFieldGroupControl('settings', 'builtin', 'fieldset')
 ```
 
 ## editorLayout.FIELDSET_CONTROL_INVALID
+
 When saving an editor layout with a field set that has `topLevelTab` control
 
 **Example:**
+
 ```javascript
 editorLayout.createFieldGroup('settings', {
   name: 'Settings'
-});
-
-editorLayout.editFieldGroup('settings')
-  .createFieldGroup('seo')
-  .name('SEO');
+})
 
-editorLayout.changeFieldGroupControl('seo', 'builtin', 'topLevelTab');
+editorLayout.editFieldGroup('settings').createFieldGroup('seo').name('SEO')
```

Comment:
I just realized that there’s quite a few unintended changes in this file. I’ll remove those.

Reply:
I can keep these changes or remove them but if I remove them we have to update the prettier rules because this rewrites the documentation every time: https://github.com/contentful/contentful-migration/blob/78e46f9f753749654c8361bc468f706b69a6d5fe/package.json#L155-L157